### ISSUE TITLE
EMD and Cosine loss functions don't seem to work 

### ISSUE THREAD

**Issue by penfoldtj** [NewcastleRSE/xray-spectroscopy-ml#36]
This in.json:

{
    "x_path": "/home/nick/XAS-3dtm/fe/preconv/xyz_train/",
    "y_path": "/home/nick/XAS-3dtm/fe/preconv/xanes_train/",

    "descriptor_type": "wacsf",
    "descriptor_params": {
        "r_min": 1.0,
        "r_max": 6.0,
        "n_g2": 32,
        "n_g4": 64,
        "z": [1,2,4,8,16,32,64,128]
    },

    "hyperparams": {
        "batch_size": 32,
        "n_hl": 2,
        "hl_ini_dim": 256,
        "hl_shrink": 0.5,
        "activation": "prelu",
        "loss": {"loss_fn" : "emd", "loss_args" : null},
        "lr": 0.00001,
        "dropout": 0.2,
        "weight_init_seed":2023,
        "kernel_init": "xavier_uniform",
        "bias_init": "zeros"
    },

    "seed": 2021,
    "epochs": 300,
    "variance_threshold": 0.0
}
